pczt/src/orchard.rs

```diff
@@ -0,0 +1,536 @@
+use std::collections::BTreeMap;
+
+#[cfg(feature = "orchard")]
+use ff::PrimeField;
+use getset::{Getters, MutGetters};
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
+
+use crate::{
+    common::Zip32Derivation,
+    roles::combiner::{merge_map, merge_optional},
+};
+
+/// PCZT fields that are specific to producing the transaction's Orchard bundle (if any).
+#[derive(Clone, Debug, Serialize, Deserialize, Getters)]
+pub struct Bundle {
+    /// The Orchard actions in this bundle.
+    ///
+    /// Entries are added by the Constructor, and modified by an Updater, IO Finalizer,
+    /// Signer, Combiner, or Spend Finalizer.
+    #[getset(get = "pub")]
+    pub(crate) actions: Vec<Action>,
+
+    /// The flags for the Orchard bundle.
+    ///
+    /// Contains:
+    /// - `enableSpendsOrchard` flag (bit 0)
+    /// - `enableOutputsOrchard` flag (bit 1)
+    /// - Reserved, zeros (bits 2..=7)
+    ///
+    /// This is set by the Creator. The Constructor MUST only add spends and outputs that
+    /// are consistent with these flags (i.e. are dummies as appropriate).
+    pub(crate) flags: u8,
+
+    /// The net value of Orchard spends minus outputs.
+    ///
+    /// This is initialized by the Creator, and updated by the Constructor as spends or
+    /// outputs are added to the PCZT. It enables per-spend and per-output values to be
+    /// redacted from the PCZT after they are no longer necessary.
+    pub(crate) value_sum: (u64, bool),
+
+    /// The Orchard anchor for this transaction.
+    ///
+    /// Set by the Creator.
+    pub(crate) anchor: [u8; 32],
```

Comment:
Why isn't the anchor optional? It only needs to be available to the Prover, I think, and there could be situations where we would want to set it after creation.